### PR TITLE
[#3331] Update buildbot try --wait to work outside of VPN.

### PR DESCRIPTION
**Pull request by adiroiban**
Scope
=====

This fix the build output not being present in --wait after moving outside of VPN

Changes
=======

Update buildbot try to use the PB (Twisted RPC) for getting the remote logs.

In the past I used a simple hack to get logs over HTTP, but we now have HTTPS which is protected.
The right way to do is to use buildbot try all the way.

This is moslty about buildbot changes which are here https://github.com/chevah/buildbot/commit/ea0c16f78a9f4ade5dc78763a81fd84786fc1895 and in the next commit

brink was cleaned of methods which are no longer used.

Now you also get the URL to the build

Buildmaster was updated to support getting the build number together with the builder name.

brink was already released

How to test
=========

reviewers: @alibotean @dumol @brunogola

Check that changes are ok

From this branch run

```
# Get latest buildbot
paver deps

# Run a complete build on a slave
paver test_remote ubuntu-1404 --wait

# Run a specific test
paver test_remote ubuntu-1404 system.test_filesystem:TestBrinkFilesystem --wait
```

once this is merged, we will need to update all other projects

### CODE REVIEW

**Comment by adiroiban** [comment]
needs-review

File: brink/qm.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def _github_token(username, password):
     authorization = user.create_authorization(
         scopes=['repo'],
         note='Chevah PQM',
-        note_url='http://build.chevah.com/waterfall')
+        note_url='https://buildbot.chevah.com')
```
Comment (dumol) [inline brink/qm.py]: Just wondering… Shouldn't the 10443 be included here too?

Comment (adiroiban) [inline brink/qm.py]: this is not advertised anywhere... is just an internal reference for GitHub... so I went with this simpler url

**Comment by dumol** [comment]
All fine here… Just a small note from me above.

**changes-approved**

**Comment by brunogola** [comment]
changes seem fine but I can't run the tests:

$ paver test_remote ubuntu-1404 --wait
---> brink.pavement_commons.test_remote
---> brink.pavement_commons.buildbot_try
ERROR: Permission to chevah/brink.git denied to brunogola.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by adiroiban** [comment]
@brunogola please try again now.

The new github permission mode make things complicated.

I gave you access to this repo.

Thanks!

**Comment by brunogola** [comment]
working now. thanks.

changes-approved

File: brink/pavement_commons.py [hunk 2]
```diff
@@ -293,8 +292,12 @@ def test_remote(args):
     for argument in args[1:]:
         if argument.startswith('--properties=') or argument == '--wait':
             arguments.append(argument)
-        elif argument.startswith('--force_'):
-            arguments.append('--properties=%s' % argument[2:])
+        elif argument.startswith('--force'):
+            argument = argument[2:].replace('-', '_')
+            if '=' not in argument:
+                # Buildot property require an explicit value.
+                argument += '=yes'
```
Comment (alibotean) [inline brink/pavement_commons.py]: Nice :smile:

**Comment by alibotean** [comment]
> ERROR: Permission to chevah/brink.git denied to alibotean.
> fatal: Could not read from remote repository.

Same issues as @bruno with access.

**Comment by alibotean** [comment]
fingers crossed :smile:

changes-approved

**Pull request closed without merging**